packages/malloy-interfaces/thrift/malloy.thrift

```diff
@@ -181,409 +162,342 @@ struct FilterExpressionType {
 // allow me to have a union whose property is also a union, since I'm compressing them
 // into an intersection type of `{__type: } & Where`. If Where is also a union, then
 // there would be two `__type` fields...
 struct FilterOperation {
-  1: required Filter filter,
+  1: required Filter filter;
 }
 
 union Filter {
-  1: required FilterStringApplication filter_string,
+  1: required FilterStringApplication filter_string;
 }
 
 struct FilterStringApplication {
-  1: required Reference field_reference,
-  2: required string filter,
+  1: required Reference field_reference;
+  2: required string filter;
 }
 
-/**
-
-stages: [
-  {ref: ff}
-  {refin: {base: ff}, {seg}}
-  {seg}
-]
-
-stages: [
-  {
-    refinements: [
-      {ref}
-      {ref}
-      {seg}
-    ]
-  }
-]
-*/
-
 struct Query {
-  1: required QueryDefinition definition,
-  2: optional list<Annotation> annotations,
+  1: required QueryDefinition definition;
+  2: optional list<Annotation> annotations;
 }
 
 union QueryDefinition {
-  1: QueryArrow arrow,
-  2: Reference query_reference,
-  3: QueryRefinement refinement,
+  1: QueryArrow arrow;
+  2: Reference query_reference;
+  3: QueryRefinement refinement;
 }
 
 union QueryArrowSource {
-  3: QueryRefinement refinement,
-  2: Reference source_reference,
+  3: QueryRefinement refinement;
+  2: Reference source_reference;
 }
 
 struct QueryArrow {
-  1: required QueryArrowSource source,
-  2: required ViewDefinition view,
+  1: required QueryArrowSource source;
+  2: required ViewDefinition view;
 }
 
 struct QueryRefinement {
-  1: required QueryDefinition base,
-  2: required ViewDefinition refinement,
+  1: required QueryDefinition base;
+  2: required ViewDefinition refining_view;
 }
 
 union ViewDefinition {
-  1: ViewArrow arrow,
-  2: Reference view_reference,
-  3: ViewRefinement refinement,
-  4: ViewSegment segment,
+  1: ViewArrow arrow;
+  2: Reference view_reference;
+  3: ViewRefinement refinement;
+  4: ViewBlock block;
 }
 
 struct ViewRefinement {
-  1: required ViewDefinition base,
-  2: required ViewDefinition refinement,
+  1: required ViewDefinition base;
+  2: required ViewDefinition refining_view;
 }
 
 struct ViewArrow {
-  1: required ViewDefinition source,
-  2: required ViewDefinition view,
+  1: required ViewDefinition source;
+  2: required ViewDefinition view;
 }
 
-struct ViewSegment {
-  1: required list<ViewOperation> operations,
+// TODO: I'm not conviced about just calling this "block"
+struct ViewBlock {
+  1: required list<ViewStatement> operations;
 }
 
 struct Reference {
-  1: required string name,
-  2: optional list<string> path,
-  3: optional list<ParameterValue> parameters,
+  1: required string name;
+  2: optional list<string> path;
+  3: optional list<ParameterValue> parameters;
 }
 
 struct ParameterValue {
-  1: required string name,
-  2: required LiteralValue value,
+  1: required string name;
+  2: required LiteralValue value;
```

Comment:
Parameters can technically be expressions in the language; someday this might become relevant in the interface, so we may want to enable that...